src/bson.rs

```diff
@@ -1062,19 +1062,30 @@ impl Display for Binary {
 
 impl Binary {
     pub(crate) fn from_extended_doc(doc: &Document) -> Option<Self> {
```

Comment:
`RawBinary` uses `is_human_readable` to determine if it should serialize to extjson or a more compact form. Since the `Bson` based serializer/deserializer can now present itself as not-human-readable, we need to add some special handling to them for when BSON types use a different serialization output.